src/data/nat/basic.lean

```diff
@@ -260,6 +260,21 @@ iff.intro eq_zero_of_mul_eq_zero (by simp [or_imp_distrib] {contextual := tt})
 @[simp] protected theorem zero_eq_mul {a b : ℕ} : 0 = a * b ↔ a = 0 ∨ b = 0 :=
 by rw [eq_comm, nat.mul_eq_zero]
 
+lemma le_mul_pos_right {m n : ℕ} (h : n > 0) : m ≤ m * n :=
+begin
+rw [← mul_one(m),mul_assoc,one_mul],
+exact mul_le_mul_of_nonneg_left (nat.succ_le_of_lt h) dec_trivial,
+end
```

Comment:
I don't like this proof --- rewriting to create the `1`, then cleaning up the `1` you didn't really mean to make.

What about:
```
begin
  conv { to_lhs, rw (by simp : m = m * 1) },
  exact mul_le_mul_left m h,
end
```
I'm sure someone will come up with something even better. :-)

Reply:
``begin``
``conv {to_lhs, rw [← one_mul(m)]},``
``exact mul_le_mul_of_nonneg_right (nat.succ_le_of_lt h) dec_trivial,``
``end``

works, I hadn't realised conv worked for inequalities, I had only been using it on equality.

Reply:
Indent inside `begin ... end` blocks by two spaces.